### DISCUSSION TITLE
How to convert SVGs into GIFs?

### DISCUSSION THREAD

**Discussion by mementobuilds** [ImageMagick/ImageMagick#3626] [Digital Image Processing] [answered]
Hi everyone, 

I'd like to convert an SVG that has a reference to a GIF inside into a final GIF. 
The SVG shows fine with the GIF in a browser, but converting the file from SVG to GIF results in only converting the first frame of the GIF. 

that's my test.svg: 

```
<svg id="TEST" data-name="TEST" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="1000" height="1000" viewBox="0 0 2000 2000">

<image width="2000" height="2000" xlink:href="input.gif"/>

</svg>
```

imagemagick command: 
`magick convert test.svg output.gif`

the result is that only the first frame gets converted into a GIF, not the whole sequence. 

Anything I can adjust with ImageMagick command to make this work?

**Comment by snibgo** [comment]
IM will convert the SVG to a single image. It won't create a series of images that correspond to the frames in the input GIF.

**Reply by mementobuilds** [reply]
Got it, thanks. are there any other cmd-line tools that can do that or would I need a completely different approach? Need that feature for batch-processing a large amount of images.

**Comment by snibgo** [comment, marked as answer]
I don't know of a tool that would do this.

A shell script could be written. Provided the SVG has only one GIF, and you don't care about timings, it could be simple:

1. Rename input.gif to temp.gif.

2. for i in the number of frames in temp.gif:

3.   magick temp.gif[i] input.gif (so input.gif has only one frame)

4.   magick test.svg frm_%i.png (so we create one output frame)

5. end for

6. magick frm_*.png out.gif (combine the output frames into a GIF)

In step 4, you need to use leading zeros, so frm_019.png will come after frm_002.png, etc.

If the SVG has multiple GIFs with different numbers of frames, the script would be more complex.

**Reply by mementobuilds** [reply]
Thanks for your answer, will give it a try.
